src/gentropy/dataset/summary_statistics.py

```diff
@@ -114,3 +117,65 @@ def exclude_region(self: SummaryStatistics, region: str) -> SummaryStatistics:
             ),
             _schema=SummaryStatistics.get_schema(),
         )
+
+    @staticmethod
+    def get_locus_sumstats(
+        session: Session,
```

Comment:
The logic of this function is datasource specific. I believe this logic should live in the `datasource` not in the `dataset`. There are 2 red flags.

- We never pass the `session` to any `Dataset` object. A `dataset` represents the data model and doesn't need to know anything about the data session.
- You need to add conditionals that are datasource specific: `if study is not None and study.startswith("FINNGEN_")`

Reply:
After discussing further with @DSuveges. This function is probably a bit of an overkill. You just need to make sure you have the summary stats object when you want to use it and filter it by chromosome and position.  Basically the majority of the logic should live upstream. If you want, you can generalise the filtering of summary statistics based on chromosome and position.

Reply:
This makes sense - so instead I should move this code into datasource specific functions? 
datasource/gwas_catalog/summary_statistics.py for example?